File: content/projects/build-shell/starters/javascript/01-print-prompt/main.js
const readline = require("readline");

// TODO: Create a readline interface for reading from stdin
// TODO: Print the prompt "$ " to stdout
// TODO: When a line is received, print the prompt again
// TODO: When the input stream closes (EOF / Ctrl+D), exit with code 0
//
// Hints:
//   - Use readline.createInterface({ input: process.stdin, output: process.stdout })
//     OR create it with just input and handle prompting yourself
//   - Use process.stdout.write('$ ') to print without a trailing newline
//   - Listen for the 'line' event to handle each line of input
//   - Listen for the 'close' event to handle EOF

File: content/projects/build-shell/starters/python/01-print-prompt/main.py
import sys


def main():
    # TODO: Implement a REPL (Read-Eval-Print Loop)
    # TODO: Print the prompt "$ " (dollar sign + space) to stdout
    # TODO: Wait for a line of input
    # TODO: After receiving input, print the prompt again
    # TODO: Exit cleanly when you receive EOF (Ctrl+D)
    #
    # Hints:
    #   - Use print("$ ", end="", flush=True) to print the prompt without a newline
    #   - The flush=True is important! Without it, the prompt may not appear
    #     when stdout is piped (not connected to a terminal)
    #   - Use input() to read a line — it raises EOFError on EOF
    #   - Wrap everything in a while True loop with a try/except for EOFError
    pass


if __name__ == "__main__":
    main()
